core/store/src/lib.rs

```diff
@@ -325,11 +324,7 @@ impl Store {
         let file = File::create(filename)?;
         let mut file = BufWriter::new(file);
         for item in self.storage.iter_raw_bytes(column) {
-            let (key, value) = item?;
-            file.write_u32::<LittleEndian>(key.len() as u32)?;
-            file.write_all(&key)?;
-            file.write_u32::<LittleEndian>(value.len() as u32)?;
-            file.write_all(&value)?;
+            borsh::to_writer(&mut file, &item?)?
```

Comment:
I personally find it a bit confusing to start relying on the fact that `borsh` coincidentally encodes this the same way as we want here. Especially considering that we don't use it when decoding values. I would rather prefer having `write_vec` method (similar to the newly added `read_vec`) to explicitly writes bytes.